experimental/fast_lossless/fast_lossless.cc

```diff
@@ -886,17 +957,156 @@ void CollectSamples(const unsigned char* rgba, size_t x0, size_t y0,
                    row_sample_collectors);
 }
 
+void PrepareDCGlobalPalette(bool is_single_group, size_t width, size_t height,
+                            const PrefixCode& code,
+                            const std::vector<uint32_t>& palette,
+
+                            size_t pcolors_real, BitWriter* output) {
+  PrepareDCGlobalCommon(is_single_group, width, height, code, output);
+  output->Write(2, 0b01);     // 1 transform
+  output->Write(2, 0b01);     // Palette
+  output->Write(5, 0b00000);  // Starting from ch 0
+  output->Write(2, 0b10);     // 4-channel palette (RGBA)
+  size_t pcolors = (pcolors_real + kChunkSize - 1) / kChunkSize * kChunkSize;
+  // pcolors > 0 and pcolors <= 1024
+  if (pcolors < 256) {
+    output->Write(2, 0b00);
+    output->Write(8, pcolors);
+  } else {
+    output->Write(2, 0b01);
```

Comment:
static_assert here that kMaxColors is <= 1024?

Reply:
done